clients/fides-js/src/fides.ts

```diff
@@ -145,20 +148,30 @@ async function init(this: FidesGlobal, providedConfig?: FidesConfig) {
     }
   }
 
-  const initialFides = getInitialFides({
-    ...config,
-    cookie: this.cookie,
-    savedConsent: this.saved_consent,
-    updateExperienceFromCookieConsent: updateExperienceFromCookieConsentNotices,
-  });
-  if (initialFides) {
+  this.experience = config.experience; // pre-fetched experience if available
+  const hasExistingCookie = !isNewFidesCookie(this.cookie);
+  const hasMigratableConsent = !!migratedConsent;
+  if (hasExistingCookie || hasMigratableConsent) {
+    /*
+     * We have enough information to initialize the Fides object before we have a valid experience.
+     * In this case, the experience is less important because the user has already consented to something.
+     * The earlier we can communicate consent to the vendor, the better.
+     */
+    const initialFides = getInitialFidesFromConsent({
```

Comment:
OK, yeah, this is clearer; basically you've hoisted the internal `if` out of the `getInitialFides` code and placed it here.